src/lib/phdi-building-blocks/phdi_building_blocks/standardize.py

```diff
@@ -1,93 +1,172 @@
 import phonenumbers
 import pycountry
 from typing import Callable, Literal, List
-
 from phdi_building_blocks.utils import find_patient_resources
+import copy
 
 
-def standardize_name(raw: str) -> str:
-    """trim spaces and force uppercase
-
-    >>> standardize_name(" JohN Doe ")
-    'JOHN DOE'
+def non_numeric_caps_standardization(raw_name: str) -> str:
     """
-
-    raw = [x for x in raw if not x.isnumeric()]
-    raw = "".join(raw)
-    raw = raw.upper()
-    raw = raw.strip()
-    return raw
+    Perform standardization on a provided name string. Name
+    standardization performs:
+    - removal of all numeric characters
+    - trimming of extra whitespace
+    - converting all characters to uppercase
+    For example, standardize_name(" JohN Doe ") is 'JOHN DOE'.
+    :param str raw: The raw string to standardize
+    :return: The standardized name string
+    :rtype: str
+    """
+    raw_name = [x for x in raw_name if not x.isnumeric()]
+    raw_name = "".join(raw_name)
+    raw_name = raw_name.upper()
+    raw_name = raw_name.strip()
+    return raw_name
 
 
-def standardize_patient_name(
-    bundle: dict, standardize: Callable = standardize_name
+def standardize_patient_names_in_bundle(
+    bundle: dict,
+    standardization_mode: str = "non_numeric_caps",
```

Comment:
I'm not a big fan of this approach for two reasons:

1. It requires the end user to familiarize themselves with our terminology, increasing the cognitive load on their end, and
2. it limits the end user to only the combination of transformations that have been pre-coded.

Instead, if we provided parameters like the ones I listed above under the `non_numeric_caps_standardization` function, then the user gets all of the control they need with minimal cognitive overhead.

What are your thoughts on that?

Reply:
I've opted to implement an approach that I think works more generally. I tried searching for a Python module that would do what we want, and I think the most robust option would be `NLTK`, but that's a pretty big module and seems like overkill for our needs. A simple homebrewed solution should suffice for now. We can re-evaluate later as our needs grow larger.